framework/topology.cpp

```diff
@@ -965,7 +967,7 @@ static void init_topology_internal(const LogicalProcessorSet &enabled_cpus)
             }
 
             pin_to_logical_processor(lp);
-            try_detection<topo_impls>(&cpu_info[i]);
+            try_detection<topo_impls>(&cpu_info[i], topology);
```

Comment:
Please run the hwloc topology detection before the pthread code and this lambda, because it doesn't need to run on the PU like this code does. Just add a topology function to the `topo_impls` list that returns true if the information has been set and therefore we shouldn't overwrite.